pkg/controller/masteruserrecord/sync_test.go

```diff
@@ -236,5 +360,37 @@ func testSyncMurStatusWithUserAccountStatus(t *testing.T, userAccount *toolchain
 		HasConditions(expMurCon).
 		HasStatusUserAccounts(test.MemberClusterName).
 		AllUserAccountsHaveStatusSyncIndex("123abc").
+		AllUserAccountsHaveCluster(toolchainv1alpha1.Cluster{
+			Name:        test.MemberClusterName,
+			APIEndpoint: "https://api.member-cluster:6433",
+			ConsoleURL:  "https://console.member-cluster/",
+		}).
 		AllUserAccountsHaveCondition(condition)
 }
+
+func newMemberCluster(cl client.Client) *cluster.FedCluster {
+	return &cluster.FedCluster{
+		Name:              test.MemberClusterName,
+		APIEndpoint:       fmt.Sprintf("https://api.%s:6433", test.MemberClusterName),
+		Client:            cl,
+		Type:              cluster.Member,
+		OperatorNamespace: test.HostOperatorNs,
```

Comment:
OperatorNamespace is a name of a namespace (in the cluster) the operator is running in - in this case it's representing member operator that is running in `test.MemberOperatorNs`
```suggestion
		OperatorNamespace: test.MemberOperatorNs,
```

Reply:
Right. But the member cluster resource exists in the Host cluster. It's used by the Host operator (which runs in the host NS) to access Member cluster.
So, the operator namespace is Host Operator NS, or I'm missing something?

The API docs could be clearer though ;)

Reply:
The host controllers need to now the member namespace - to GET/POST/UPDATE resources inside of the member cluster. To identify the member namespace we use the label `namespace` and store it as `OperatorNamespace` inside of the `FedCluster` type. The idea is that the `FedCluster` represents some cluster (either host or member) and the `OperatorNamespace` is saying which namespace the toochain operator is running in. So here the `FedCluster` represents a member cluster so it should contain  `test.MemberOperatorNs`. There is no point in storing the `test.HostOperatorNs` as the host controllers know the namespace they are running in.

I'll try to improve the docs ;-)

Reply:
That makes sense :)
Maybe targetNamespace would be a better name... but we don’t have to change the field name now. But let’s improve the docs at least.
Also I wonder why tests pass...

Reply:
@MatousJobanek could you please make sure it’s covered by tests if it’s not too tricky?